File: Python/643. Maximum Average Subarray I.py
class Solution:
    def findMaxAverage(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: float
        """
        ans = s = sum(nums[:k])
        for i in range(k, len(nums)):
            s = s - nums[i-k] + nums[i]
            ans = max(ans, s)
        return ans / k

File: C++/643. Maximum Average Subarray I.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int ans = 0,len = nums.size();
        for(int i = 0;i<k;i++)
            ans += nums[i];
        int sum = ans;
        for(int i=k;i<len;i++)
        {
            sum = sum+nums[i] - nums[i-k];
            ans = max(sum,ans);
        }
        return 1.0*ans/k;
    }
};